impeller/geometry/color.h

```diff
@@ -8,12 +8,55 @@
 #include <array>
 #include <cstdlib>
 #include <ostream>
-
 #include "impeller/geometry/scalar.h"
 
 namespace impeller {
 
 struct ColorHSB;
+struct Vector4;
+
+/// All blend modes assume that both the source (fragment output) and
+/// destination (first color attachment) have colors with premultiplied alpha.
+enum class BlendMode {
+  // The following blend modes are able to be used as pipeline blend modes or
+  // via `BlendFilterContents`.
+  kClear,
+  kSource,
+  kDestination,
+  kSourceOver,
+  kDestinationOver,
+  kSourceIn,
+  kDestinationIn,
+  kSourceOut,
+  kDestinationOut,
+  kSourceATop,
+  kDestinationATop,
+  kXor,
+  kPlus,
+  kModulate,
+
+  // The following blend modes use equations that are not available for
+  // pipelines on most graphics devices without extensions, and so they are
+  // only able to be used via `BlendFilterContents`.
+  kScreen,
+  kOverlay,
+  kDarken,
+  kLighten,
+  kColorDodge,
+  kColorBurn,
+  kHardLight,
+  kSoftLight,
+  kDifference,
+  kExclusion,
+  kMultiply,
+  kHue,
+  kSaturation,
+  kColor,
+  kLuminosity,
+
+  kLastPipelineBlendMode = kModulate,
+  kLastAdvancedBlendMode = kLuminosity,
```

Comment:
Please remove these in favor of the new values in the Entity namespace.

Reply:
done